pkg/lifecycle/binary.go

```diff
@@ -132,32 +133,52 @@ func DownloadBinary(url, destDir, destFile string, logger *zap.Logger) (string,
 		}
 	}()
 
-	logger.Info("downloading binary", zap.String("url", url))
+	logger.Info("downloading binary", zap.String("url", sourceURL))
 
-	req, err := http.NewRequest("GET", url, nil)
+	u, err := url.Parse(sourceURL)
 	if err != nil {
-		return "", fmt.Errorf("could not create request (%w)", err)
+		return "", fmt.Errorf("could not parse URL %s (%w)", sourceURL, err)
 	}
-	client := &http.Client{}
-	resp, err := client.Do(req)
-	if err != nil {
-		return "", fmt.Errorf("HTTP GET %s failed (%w)", url, err)
-	}
-	defer resp.Body.Close()
 
-	if resp.StatusCode != http.StatusOK {
-		return "", fmt.Errorf("HTTP GET %s failed with error %d", url, resp.StatusCode)
-	}
+	switch u.Scheme {
+	case "http", "https":
```

Comment:
This makes the test easier since I don't want to mock the HTTP server. The validation logic in the piped configuration remains to only allow the "https" and "file" schemes.

Reply:
📝 It is difficult to test with https server because `DownloadBinary` uses http.Client{} internally and can't inject for now. So we can't use httptest.NewTLSServer.

Reply:
@t-kikuc For the question you asked 🙏

Reply:
@khanhtc1202 

I got it.
binary.go allows http and https, `PipedPlugin` disallows http.

https://github.com/pipe-cd/pipecd/blob/dcecc2d9b88201fa5228350411ae5895cbb44547/pkg/configv1/piped.go#L1328-L1330